src/main/java/com/epam/flightscontroltool/control/service/AirplaneService.java

```diff
@@ -0,0 +1,92 @@
+package com.epam.flightscontroltool.control.service;
+
+import com.epam.flightscontroltool.boundary.dto.AirplaneDto;
+import com.epam.flightscontroltool.boundary.exception.AirplaneNotFoundException;
+import com.epam.flightscontroltool.boundary.exception.CompanyNotFoundException;
+import com.epam.flightscontroltool.boundary.request.AirplaneRequest;
+import com.epam.flightscontroltool.control.repository.AirplaneRepository;
+import com.epam.flightscontroltool.control.repository.CompanyRepository;
+import com.epam.flightscontroltool.entity.Airplane;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class AirplaneService {
+
+    private final AirplaneRepository airplaneRepository;
+    private final CompanyRepository companyRepository;
+
+    public Page<AirplaneDto> getAirplanes(Long companyId, int pageNumber, int pageSize) {
+        return (companyId == null)
+                ? getAirplanes(pageNumber, pageSize)
+                : getAirplanesByCompanyId(companyId, pageNumber, pageSize);
+    }
+
+    public Page<AirplaneDto> getAirplanes(int pageNumber, int pageSize) {
+        var pageRequest = PageRequest.of(pageNumber, pageSize);
+        var page = this.airplaneRepository.findAll(pageRequest);
+        return page.map(AirplaneDto::from);
+    }
+
+    public Page<AirplaneDto> getAirplanesByCompanyId(Long companyId, int pageNumber, int pageSize) {
+        var pageRequest = PageRequest.of(pageNumber, pageSize);
+        var page = this.airplaneRepository.getAllByCompanyId(companyId, pageRequest);
+        return page.map(AirplaneDto::from);
+    }
+
+    @Transactional
+    public AirplaneDto create(AirplaneRequest airplaneRequest) {
```

Comment:
I suggest using only business entities (Airplane, Airport, City) in the service layer. The main idea of layered architecture is that each layer only knows about one layer beneath it, and doesn't know anything about layers above it. With the current implementation, the business layer (Services) know everything about actual data representation on our endpoints and how to trsnform data to this format.
Imagine the situation when we need to handle another set of API (like import from excel spreadsheets) in that case we cannot reuse this class methods at all as it only works with representation layer objects. It's essentially become a presentation layer itself.